tests/test_writer.py

```diff
@@ -1,27 +1,41 @@
 from pytest import fixture, mark, raises
 
 from piptools._compat import FormatControl
+from piptools.scripts.compile import cli
 from piptools.utils import comment
 from piptools.writer import OutputWriter
 
 
 @fixture
-def writer(tmp_file):
-    return OutputWriter(
-        src_files=["src_file", "src_file2"],
-        dst_file=tmp_file,
-        dry_run=True,
-        emit_header=True,
-        emit_index=True,
-        emit_trusted_host=True,
-        annotate=True,
-        generate_hashes=False,
-        default_index_url=None,
-        index_urls=[],
-        trusted_hosts=[],
-        format_control=FormatControl(set(), set()),
-        allow_unsafe=False,
-    )
+def writer(tmpdir):
+    with open("src_file", "w"):
+        pass
+
+    with open("src_file2", "w"):
+        pass
```

Comment:
Could be nested (to save some lines).

Reply:
Right! Thanks.

Reply:
src_file and src_file2 should be in the tmp_file dir

Reply:
Right, they are already in the temp dir, see `runner` fixture. It's running under isolated file system.